### PR TITLE
[Refact](type system) refact column with arrow serde

### PR DESCRIPTION
**Pull request by amorynan**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/data_types/serde/data_type_string_serde.cpp [hunk 1]
```diff
@@ -63,5 +66,62 @@ void DataTypeStringSerDe::read_one_cell_from_jsonb(IColumn& column, const JsonbV
...
 }
+
+void DataTypeStringSerDe::write_column_to_arrow(const IColumn& column, const UInt8* null_map,
+                                                arrow::ArrayBuilder* array_builder, int start,
+                                                int end) const {
+    const auto& string_column = assert_cast<const ColumnString&>(column);
+    auto& builder = assert_cast<arrow::StringBuilder&>(*array_builder);
+    for (size_t string_i = start; string_i < end; ++string_i) {
+        if (null_map && null_map[string_i]) {
+            checkArrowStatus(builder.AppendNull(), column.get_name(),
```
Comment (yiguolei) [inline be/src/vec/data_types/serde/data_type_string_serde.cpp:77]: method name should be   check_arrow_status

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.69 seconds
 stream load tsv:          430 seconds loaded 74807831229 Bytes, about 165 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230504034645_clickbench_pr_137915.html

**Pull request merged**